pkg/serviceaccount/legacy.go

```diff
@@ -126,6 +140,30 @@ func (v *legacyValidator) Validate(ctx context.Context, tokenData string, public
 			klog.V(4).Infof("Service account UID no longer matches %s/%s: %q != %q", namespace, serviceAccountName, string(serviceAccount.UID), serviceAccountUID)
 			return nil, fmt.Errorf("ServiceAccount UID (%s) does not match claim (%s)", serviceAccount.UID, serviceAccountUID)
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.LegacyServiceAccountTokenTracking) {
+			for _, ref := range serviceAccount.Secrets {
+				if ref.Name == secret.Name {
+					warning.AddWarning(ctx, "", "Use tokens from the TokenRequest API or manually created secret-based tokens instead of auto-generated secret-based tokens.")
+					break
+				}
+			}
+			now := time.Now().UTC()
+			yesterday := now.AddDate(0, 0, -1).Format("2006-01-02")
```

Comment:
I think we want to no-op if the existing value is today or *tomorrow*. That ensures we update the secret as soon as possible, and a slightly older skewed API server won't regress us.

Reply:
ah, i was concerned that this will skip update. only the lagged behind api server should skip the update.